docs/upgrade.md

```diff
@@ -0,0 +1,58 @@
+# Upgrade
+
+Currently the upgrade can be performed just between follow up downstream builds
+e.g. 171 -> 172.
+There is opened [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1767400) which
+is tracking the issue which prevents us to upgrade to the latest available build.
+
+There is prepared config option `upgrade_to_latest` in the DEPLOYMENT section
+which can be set or will be the default value when the BZ gets resolved.
+
+## Deployment
+
+To be able to upgrade your cluster you cannot deploy the latest downstream
+build, because then you won't have any next build available for upgrade.
+
+For this purpose we have this config file: [upgrade.yaml](/conf/ocsci/upgrade.yaml)
+which you can pass via `--ocsci-conf` when you are running the deployment. This will
+automatically take not latest build but one before.
+
+## Upgrade execution
+
+For upgrade we have few pytest marks defined [here](/ocs_ci/framework/pytest_customization/marks.py):
+
+* **before_upgrade** - mark tests which are suppose to run before upgrade
+* **upgrade** - this is mark for upgrade itself
+* **after_upgrade** - mark tests which are suppose to run after upgrade
+
+Those marks has to be imported from mentioned
+[module](/ocs_ci/framework/pytest_customization/marks.py) and your test have to
+be decorated by those marks as we combine those mark decorators with order
+decorator used by `pytest-ordering` plugin to properly order the tests.
+
+To execute the upgrade you can use this command:
+
+```bash
+run-ci tests/
+    --cluster-name kerberos_ID-ocs-deployment \
+    --cluster-path /home/my_user/my-ocs-dir \
+    -m upgrade
+ ```
+
+If you would like to trigger also pre/post upgrade tests run with following
+markers:
+
+```bash
+run-ci tests/
+    --cluster-name kerberos_ID-ocs-deployment \
+    --cluster-path /home/my_user/my-ocs-dir \
+    -m 'before_upgrade or upgrade or after_upgrade'
+ ```
+
+## TODO
+
+Once there will be 4.3 build and we will have different branch for OCS-CI we
+will need to run upgrade in two steps. First run just with `-m before_upgrade`
+from 4.2 branch if we will have such tests and then with
+`-m 'upgrade or after_upgrade'` from 4.3 (master branch) to run everything from
+proper branches designed for version of OCS.
```

Comment:
This TODO section needs some update based on our yesterday discussion about not using branches right now.

Reply:
please update the doc to reflect the changes.

Reply:
@petr-balogh as already discussed offline, we would also need to fulfil all the Entry and exit criteria for each upgrade scenario. Once that is dome, only then can we run the tier1 test cases on the cluster.

We would be coming up with a plan on how to combine these validations with the actual upgrade process(as covered in this PR)

Reply:
I am currently depending on other changes in the deployment like this PR: https://github.com/red-hat-storage/ocs-ci/pull/983  and I am working on some changes with local storage operator validation but cause of this bug https://bugzilla.redhat.com/show_bug.cgi?id=1770183 I am unable to check the version of this operator so I will have to workaround this. Also there are issues with deployment since last week so it's problematic with verification now. Once I have changes verified locally I will push to this PR!

@nehaberry the tier1 execution on upgraded environment will be done in second execution of ocs-ci on the environment after the upgrade.